Spigot-Server-Patches/0402-Fix-NPE-when-setting-a-player-s-spectator-target.patch

```diff
@@ -0,0 +1,215 @@
+From 7ac07ac07ac07ac07ac07ac07ac07ac07ac07ac0 Mon Sep 17 00:00:00 2001
+From: Caleb Bassham <caleb.bassham@gmail.com>
+Date: Sat, 20 Oct 2018 12:56:56 -0500
+Subject: [PATCH] Fix NPE when setting a player's spectator target
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityPlayer.java b/src/main/java/net/minecraft/server/EntityPlayer.java
+index 7ac07ac07ac0..7ac07ac07ac0 100644
+--- a/src/main/java/net/minecraft/server/EntityPlayer.java
++++ b/src/main/java/net/minecraft/server/EntityPlayer.java
+@@ -1,24 +1,11 @@
+ package net.minecraft.server;
+ 
++import com.google.common.base.Preconditions;
+ import com.google.common.collect.Lists;
+ import com.mojang.authlib.GameProfile;
+ import io.netty.buffer.Unpooled;
+-import io.netty.util.concurrent.Future;
+-import io.netty.util.concurrent.GenericFutureListener;
+-import java.util.ArrayDeque; // Paper
+-import java.util.ArrayList;
+-import java.util.Collection;
+-import java.util.Deque; // Paper
+-import java.util.Iterator;
+-import java.util.List;
+-import java.util.Random;
+-import java.util.function.Consumer;
+-import javax.annotation.Nullable;
+ import org.apache.logging.log4j.LogManager;
+ import org.apache.logging.log4j.Logger;
+-
+-// CraftBukkit start
+-import com.google.common.base.Preconditions;
+ import org.bukkit.Bukkit;
+ import org.bukkit.GameMode;
+ import org.bukkit.Location;
+@@ -33,6 +20,11 @@ import org.bukkit.event.player.PlayerGameModeChangeEvent;
+ import org.bukkit.event.player.PlayerLocaleChangeEvent;
+ import org.bukkit.event.player.PlayerTeleportEvent.TeleportCause;
+ import org.bukkit.inventory.MainHand;
++
++import javax.annotation.Nullable;
++import java.util.*;
++
++// CraftBukkit start
+ // CraftBukkit end
+ 
+ public class EntityPlayer extends EntityHuman implements ICrafting {
+@@ -62,9 +54,19 @@ public class EntityPlayer extends EntityHuman implements ICrafting {
+     private EntityHuman.EnumChatVisibility cs;
+     private boolean ct = true;
+     private long cu = SystemUtils.b();
+-    private Entity cv; private void setSpectatorTargetField(Entity e) { this.cv = e; } // Paper - OBFHELPER
++    private Entity cv;
++
++    private void setSpectatorTargetField(Entity e) {
++        this.cv = e;
++    } // Paper - OBFHELPER
```

Comment:
Avoid formatting obfuscation helpers, they should be in the same line as the field the OBFHelper is going to modify.

Reply:
Yea I am already fixing it right now sorry. IntelliJ formatted it and I didn't realize.